core/src/test/java/feast/core/grpc/CoreServiceImplTest.java

```diff
@@ -0,0 +1,135 @@
+package feast.core.grpc;
+
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import feast.core.CoreServiceGrpc;
+import feast.core.CoreServiceGrpc.CoreServiceBlockingStub;
+import feast.core.CoreServiceProto.CoreServiceTypes.GetUploadUrlRequest;
+import feast.core.CoreServiceProto.CoreServiceTypes.GetUploadUrlRequest.FileType;
+import feast.core.CoreServiceProto.CoreServiceTypes.GetUploadUrlResponse;
+import feast.core.CoreServiceProto.CoreServiceTypes.GetUploadUrlResponse.HttpMethod;
+import feast.core.service.JobManagementService;
+import io.grpc.inprocess.InProcessChannelBuilder;
+import io.grpc.inprocess.InProcessServerBuilder;
+import io.grpc.testing.GrpcCleanupRule;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.junit.Assume.assumeTrue;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CoreServiceImplTest {
+  @Rule public final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
+  @InjectMocks private CoreServiceImpl coreService = new CoreServiceImpl();
+
+  @Mock private JobManagementService jobManagementService;
+
+  @Test
+  public void getUploadUrl() throws IOException {
+    // Skip this test if GOOGLE_APPLICATION_CREDENTIALS is not set
+    // because this test requires Google service account key and connection to "actual"
+    // Google Cloud Storage service
+    assumeTrue(StringUtils.isNotEmpty(System.getenv("GOOGLE_APPLICATION_CREDENTIALS")));
+
+    // Mock the workspace output from jobManagementService to a bucket we can use for testing
+    String bucketName = "feast-templocation-kf-feast";
```

Comment:
Shouldn't this be parameterized so that the test runner can run for people without access to this specific bucket?

Reply:
It's parameterized now with `-Dtestbucket=mybucket` when running the test